### DISCUSSION TITLE
No connection after radio module change

### DISCUSSION THREAD

**Discussion by morph027** [tbnobody/OpenDTU#708] [General]
I was using a cheap module from AZ which works, but sometimes connection get's lost for some periods.

I'm using default ESP32 with the https://github.com/dokuhn/openDTU-BreakoutBoard

Tried to upgrade to the [ebyte modules](https://www.amazon.de/gp/product/B07P8TTHZH/?th=1) which were recommended in some issues and discussions.

Module gets recognized by the system but does not fetch any data from the inverter.

![Screenshot from 2023-03-17 11-39-31](https://user-images.githubusercontent.com/600106/225884365-df60c052-677c-49dc-bfc6-58675e7d20b6.png)

Console logs of the ebyte module

```
11:38:47.329 > Fetch inverter: xxxx
11:38:47.329 > Request SystemConfigPara
11:38:47.376 > TX RealTimeRunData Channel: 23 --> 15 82 96 80 37 80 14 73 48 80 0B 00 64 14 43 37 00 00 00 00 00 00 00 00 9F 11 18 
11:38:47.622 > RX Period End
11:38:47.622 > All missing
11:38:47.622 > Nothing received, resend whole request
11:38:47.622 > TX RealTimeRunData Channel: 40 --> 15 82 96 80 37 80 14 73 48 80 0B 00 64 14 43 37 00 00 00 00 00 00 00 00 9F 11 18 
11:38:47.867 > RX Period End
11:38:47.867 > All missing
11:38:47.867 > Nothing received, resend whole request
11:38:47.867 > TX RealTimeRunData Channel: 61 --> 15 82 96 80 37 80 14 73 48 80 0B 00 64 14 43 37 00 00 00 00 00 00 00 00 9F 11 18 
11:38:48.111 > RX Period End
11:38:48.111 > All missing
11:38:48.111 > Nothing received, resend whole request
11:38:48.111 > TX RealTimeRunData Channel: 75 --> 15 82 96 80 37 80 14 73 48 80 0B 00 64 14 43 37 00 00 00 00 00 00 00 00 9F 11 18 
11:38:48.356 > RX Period End
11:38:48.356 > All missing
11:38:48.356 > Nothing received, resend whole request
11:38:48.356 > TX RealTimeRunData Channel: 3 --> 15 82 96 80 37 80 14 73 48 80 0B 00 64 14 43 37 00 00 00 00 00 00 00 00 9F 11 18 
11:38:48.559 > RX Period End
11:38:48.559 > All missing
11:38:48.559 > Nothing received, resend count exeeded
11:38:48.605 > TX AlarmData Channel: 23 --> 15 82 96 80 37 80 14 73 48 80 11 00 64 14 43 37 00 00 00 00 00 00 00 00 45 0A C3 
11:38:49.251 > RX Period End
11:38:49.251 > All missing
11:38:49.251 > Nothing received, resend whole request
11:38:49.251 > TX AlarmData Channel: 40 --> 15 82 96 80 37 80 14 73 48 80 11 00 64 14 43 37 00 00 00 00 00 00 00 00 45 0A C3 
11:38:49.895 > RX Period End
11:38:49.895 > All missing
11:38:49.895 > Nothing received, resend whole request
11:38:49.895 > TX AlarmData Channel: 61 --> 15 82 96 80 37 80 14 73 48 80 11 00 64 14 43 37 00 00 00 00 00 00 00 00 45 0A C3 
11:38:50.539 > RX Period End
11:38:50.539 > All missing
11:38:50.539 > Nothing received, resend whole request
11:38:50.539 > TX AlarmData Channel: 75 --> 15 82 96 80 37 80 14 73 48 80 11 00 64 14 43 37 00 00 00 00 00 00 00 00 45 0A C3 
11:38:51.183 > RX Period End
11:38:51.183 > All missing
11:38:51.183 > Nothing received, resend whole request
11:38:51.183 > TX AlarmData Channel: 3 --> 15 82 96 80 37 80 14 73 48 80 11 00 64 14 43 37 00 00 00 00 00 00 00 00 45 0A C3 
11:38:51.786 > RX Period End
11:38:51.786 > All missing
11:38:51.786 > Nothing received, resend count exeeded
11:38:51.833 > TX SystemConfigPara Channel: 23 --> 15 82 96 80 37 80 14 73 48 80 05 00 64 14 43 37 00 00 00 00 00 00 00 00 51 1E D7 
11:38:52.079 > RX Period End
11:38:52.079 > All missing
11:38:52.079 > Nothing received, resend whole request
11:38:52.079 > TX SystemConfigPara Channel: 40 --> 15 82 96 80 37 80 14 73 48 80 05 00 64 14 43 37 00 00 00 00 00 00 00 00 51 1E D7 
11:38:52.323 > RX Period End
11:38:52.323 > All missing
11:38:52.323 > Nothing received, resend whole request
11:38:52.323 > TX SystemConfigPara Channel: 61 --> 15 82 96 80 37 80 14 73 48 80 05 00 64 14 43 37 00 00 00 00 00 00 00 00 51 1E D7 
11:38:52.569 > RX Period End
11:38:52.569 > All missing
11:38:52.569 > Nothing received, resend whole request
11:38:52.569 > TX SystemConfigPara Channel: 75 --> 15 82 96 80 37 80 14 73 48 80 05 00 64 14 43 37 00 00 00 00 00 00 00 00 51 1E D7 
11:38:52.812 > RX Period End
11:38:52.812 > All missing
11:38:52.812 > Nothing received, resend whole request
11:38:52.812 > TX SystemConfigPara Channel: 3 --> 15 82 96 80 37 80 14 73 48 80 05 00 64 14 43 37 00 00 00 00 00 00 00 00 51 1E D7 
11:38:53.044 > RX Period End
11:38:53.044 > All missing
11:38:53.044 > Nothing received, resend count exeeded
```

**Comment by alexandertonino** [comment]
I had the same problem.
Check if the CE signal is connected correctly. 

The ebyte module works a lot better for me.

**Reply by morph027** [reply]
You mean by checking the soldering? The other module works. Board is not wired as i'm using the https://github.com/dokuhn/openDTU-BreakoutBoard

**Reply by alexandertonino** [reply]
well.. yes.. the connection between ESP32 Gpio4 and the CE pin of the module.

**Reply by morph027** [reply]
Wired up all connections according to the README and it just does not work. Weird. Chance of getting 2 broken modules out of 2? :see_no_evil:

**Reply by morph027** [reply]
Okay, another test went wrong. Doublechecked all cables, tried another ESP32 module, nothing works. Guess it's the modules itselves. Just wanted to make sure before returning to the vendor.

**Comment by morph027** [comment]
Hmpf, weird, just ordered another batch of 2 modules and none seems to work. Can't imagine this as the modules are recommended a lot here. Any hints what and how to triple check the connection?

I already searched for the datasheet and confirmed that the pinout is the same as for all other modules out there.

**Comment by Solarteur** [comment]
So many potentially defective nRF24L01+ modules seem strange.
Have you checked the soldered connections of the socket strips for ESP32 and nRF24L01+ and resoldered them if necessary?
Have you tested a different transmit power (PA Level) for the nRF24L01+ module?
Did you solder the electrolytic capacitor (10 - 22 µF)?
Have you tried a different power supply/cable?

**Reply by morph027** [reply]
> So many potentially defective nRF24L01+ modules seem strange.

Yes, seems strange.

> Have you checked the soldered connections of the socket strips for ESP32 and nRF24L01+ and resoldered them if necessary?

I've tested without the sockets using just wires and even tested another ESP32. The AZ Delivery modules are working fine on the same board/wiring

> Have you tested a different transmit power (PA Level) for the nRF24L01+ module?

Yes.

> Did you solder the electrolytic capacitor (10 - 22 µF)?

Yes

> Have you tried a different power supply/cable?

Will try that. Does the ebyte module draw significant more power?

**Reply by Solarteur** [reply]
The power requirement of nRF24L01+ modules essentially depends on whether a module with a printed antenna or a module with an amplifier and external antenna is used.
I know that some of my suggestions are far-fetched, but I didn't want to leave out any possible mistakes. Ultimately, many OpenDTU work without any problems.

**Reply by morph027** [reply]
I appreciate your help. I feel the same, chance that 4 modules are broken is quite low.

Will try to solder another set.

It really bugs me, that the cheap [module](https://www.amazon.de/dp/B06XJN417D/ref=twister_B0822HLCDJ?_encoding=UTF8&psc=1) with printed antenna just works, even with PA level set to maximum (basement with steel doors and ferroconcrete ceilings) on the same circuit and board.

**Reply by DPO99** [reply]
I got problems with the "big" one too. I solved this problem with using [this ](https://www.makershop.de/module/funk/nrf24l01-adapter/)and connected it to the 5v Power from the ESP.
